src/containers/unit_list.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { selectUnit } from '../actions/action_select_unit'
+import { bindActionCreators } from 'redux'
+import PropTypes from 'prop-types'
+
+/**
+ * container UnitList:
+ * represents a list of search result units
+ */
+class UnitList extends Component {
+  constructor (props) {
+    super(props)
+
+    this.renderList = this.renderList.bind(this)
+  }
+
+  // @returns unit list item
+  renderList () {
+    return (
+      // map an array of units to unit list items
+      this.props.searchResultUnits.map((unit) => {
+        return (
+          <li
+            key={unit.unitCode}
+            onClick={() => this.props.selectUnit(unit)}
+            className="list-group-item">{unit.unitCode} - {unit.unitName}</li>
+        )
+      })
+    )
+  }
+
+  render () {
+    // when the search hasn't been kicked off
+    if (!this.props.searchResultUnits) {
```

Comment:
this area can be dramatically shortened an made easier to use using [tenary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator). 

{
  !this.props.searchResultUnits &&
  <div> Search for a unit to get started </div>
}

You're repeating alot of variables, use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), i.e. 
```javascript
render(){
   const { serachResultUnits } = this.props
   return (
      {
       searchResultUnits.length === 0)  && <div > No units found </div>
      }
   )
}
```